### PR TITLE
[BFUN-2729] First pass at Toggl API v9

### PR DESCRIPTION
**Pull request by rtrochet**
BFUN-2729: Updates `togglwrapper` to handle v9 endpoints. 

This is a major change that breaks almost all the old endpoints. https://github.com/Zapgram/pilot/pull/19038 points at this branch, and changes there will handle changes in output, which this library doesn't care about since we return the raw JSON, whatever shape it has.

The tests in this module pass.

If I could make more changes, I'd lint these files, type them, and get rid of the mixins. But Mike has convinced me that it's better to make the smallest possible set of changes in hopes of getting them accepted upstream.

Even though these are the smallest possible changes, this PR is pretty intense and I'm happy to do a pair review. The commits here are meaningless :(

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -23,13 +21,12 @@
 
 setup(
     name='togglwrapper',
-    version=__version__,
```
Comment (rtrochet) [inline setup.py:26]: This set of imports broke dev, so I hardcoded the next major version.

File: togglwrapper/api.py [hunk 2]
```diff
@@ -3,53 +3,73 @@
 """
 togglwrapper.api
 """
-
+from __future__ import annotations
```
Comment (rtrochet) [inline togglwrapper/api.py]: I can back this out if you want.

File: tests.py [hunk 3]
```diff
@@ -747,5 +770,5 @@ def test_delete(self):
 
 
 if __name__ == '__main__' and __package__ is None:
-    __package__ = "toggl"
+    __package__ = 'toggl'
```
Comment (rtrochet) [inline tests.py:772]: Gah, I see why people use formatters—the inconsistency here is a little maddening (considering how hard I tried to back out my formatting changes).

File: togglwrapper/api.py [hunk 4]
```diff
@@ -3,53 +3,73 @@
...
+        if self.uri is None or self.prepend_workspace_id is None:
             # Helper error for subclasses that forget to specify their own URI
-            raise NotImplementedError('Must define a URI.')
+            raise NotImplementedError(
+                'Must define a URI and whether to prepend the workspace ID.'
+            )
 
-    @classmethod
-    def _compile_uri(cls, id=None, ids=None, child_uri=None):
+    def _compile_uri(self, id=None, ids=None, child_uri=None, parent_uri=None):
```
Comment (rtrochet) [inline togglwrapper/api.py:41]: Big change: this is no longer a classmethod so I can access the workspace_id.

File: togglwrapper/decorators.py [hunk 5]
```diff
@@ -9,21 +9,25 @@
...
 from .exceptions import AuthError
 
 
 def return_json(func):
-    """ Returns the JSON content of a requests.Response. """
+    """Returns the JSON content of a requests.Response."""
+
     @wraps(func)
-    def inner(*args, **kwargs):
+    def inner(*args, **kwargs) -> Any:
```
Comment (rtrochet) [inline togglwrapper/decorators.py]: Back out this typing.

**Review by pilot-jasonslchen — APPROVED** [review]
good to ship once you've backed out what you wanted to back out!

**Pull request merged**